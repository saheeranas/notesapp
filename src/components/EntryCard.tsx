import React from 'react';
import {StyleSheet, View, Pressable} from 'react-native';

import {Text} from '@ui-kitten/components';
import dayjs from 'dayjs';

interface EntryCardProps {
  item: any;
  onPress: () => void;
}

const EntryCard: React.FC<EntryCardProps> = ({item: {desc, date}, onPress}) => {
  const dayStr = dayjs(date).format('MMM D h:mm A');
  return (
    <Pressable style={styles.listItem} onPress={onPress}>
      <View style={styles.listItemInner}>
        <Text style={styles.desc}>{desc.substr(0, 50)}</Text>
        <View style={styles.dateWrp}>
          <Text style={styles.dayStr}>{dayStr}</Text>
        </View>
      </View>
    </Pressable>
  );
};

export default EntryCard;

const styles = StyleSheet.create({
  listItem: {
    marginBottom: 10,
    width: '48%',
  },
  listItemInner: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: '#fff',
    borderRadius: 5,
    alignItems: 'flex-start',
  },
  dateWrp: {
    marginBottom: 3,
    alignItems: 'center',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: '#ccc',
    paddingTop: 8,
  },
  dayStr: {
    fontSize: 10,
    color: '#333333',
    textAlign: 'center',
    textTransform: 'uppercase',
  },
  desc: {
    fontSize: 15,
    color: '#333333',
    flex: 1,
    maxHeight: 40,
    paddingBottom: 8,
  },
});
