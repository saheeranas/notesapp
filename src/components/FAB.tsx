import React from 'react';
import {StyleSheet} from 'react-native';
import {Button, Icon, ButtonProps} from '@ui-kitten/components';

interface FABProps extends ButtonProps {
  onPress: () => void;
}

const AddIcon = () => (
  <Icon name="plus-outline" style={styles.icon} fill="#fff" />
);

const FAB = (props: FABProps) => {
  return (
    <Button
      style={styles.fab}
      {...props}
      onPress={() => props.onPress()}
      accessoryLeft={AddIcon}
      status="warning"
      testID="fab_button"
    />
  );
};

const styles = StyleSheet.create({
  fab: {
    width: 46,
    height: 46,
    borderRadius: 46 / 2,
    borderWidth: 0,
    position: 'absolute',
    elevation: 1,
    zIndex: 1,
    bottom: 16,
    right: 16,
    padding: 0,
  },
  icon: {width: 24, height: 24},
});

export default FAB;
