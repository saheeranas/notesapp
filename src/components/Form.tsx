import React, {forwardRef} from 'react';
import {TextInput, StyleSheet, View} from 'react-native';
import {Input, InputProps} from '@ui-kitten/components';

interface SearchProps extends InputProps {}

export const TextArea = forwardRef((props: any, ref) => {
  return <TextInput ref={ref} style={styles.textarea} {...props} />;
});

export const Search = (props: SearchProps) => {
  return (
    <View style={styles.searchCard}>
      <Input style={styles.searchInput} {...props} />
    </View>
  );
};

const styles = StyleSheet.create({
  textarea: {
    borderWidth: 0,
  },
  // Search
  searchCard: {
    backgroundColor: 'transparent',
    paddingHorizontal: 16,
    paddingTop: 0,
    paddingBottom: 3,
  },
  searchInput: {
    backgroundColor: '#ececec',
  },
});
