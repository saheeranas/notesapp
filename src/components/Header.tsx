import React from 'react';
import {StyleSheet} from 'react-native';
import {
  Icon,
  TopNavigation,
  TopNavigationAction,
  Text,
} from '@ui-kitten/components';

import {HeaderType} from '../types/types';

const BackIcon = () => (
  <Icon style={{width: 20, height: 20}} name="arrow-back" fill="#000" />
);

const KebabIcon = () => (
  <Icon
    style={{width: 20, height: 20}}
    name="more-vertical-outline"
    fill="#000"
  />
);

const Header: React.FC<HeaderType> = ({
  title = 'Diary',
  navigation,
  hideBack = false,
  style,
  onPressMenu,
}) => {
  const BackAction = () => (
    <TopNavigationAction icon={BackIcon} onPress={() => navigation.goBack()} />
  );

  const MenuButton = () => (
    <TopNavigationAction
      icon={KebabIcon}
      onPress={() => {
        onPressMenu && onPressMenu();
      }}
    />
  );

  const Title = () => <Text category="s1">{title}</Text>;

  return (
    <TopNavigation
      accessoryLeft={hideBack ? undefined : BackAction}
      accessoryRight={onPressMenu ? MenuButton : undefined}
      title={Title}
      alignment="center"
      style={[styles.header, style]}
    />
  );
};

const styles = StyleSheet.create({
  header: {
    backgroundColor: 'transparent',
  },
});

export default Header;
