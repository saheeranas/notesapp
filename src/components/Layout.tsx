import React, {ReactNode} from 'react';
import {StyleSheet} from 'react-native';
import {Layout as UILayout} from '@ui-kitten/components';

interface LayoutProps {
  children?: ReactNode;
}

export const Layout = ({children}: LayoutProps) => {
  return <UILayout style={styles.layout}>{children}</UILayout>;
};

const styles = StyleSheet.create({
  safearea: {
    flex: 1,
  },
  layout: {
    position: 'relative',
    flex: 1,
    backgroundColor: '#F4F4F4',
  },
});
