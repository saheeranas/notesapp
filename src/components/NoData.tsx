import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Text} from '@ui-kitten/components';

interface NoDataProps {
  title?: string;
}

const NoData = ({title = 'No Data'}: NoDataProps) => {
  return (
    <View style={styles.wrapper}>
      <Text status="basic" category="p2">
        {title}
      </Text>
    </View>
  );
};

export default NoData;

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
