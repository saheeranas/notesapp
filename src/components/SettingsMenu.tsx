import React from 'react';
import {StyleSheet, View, TouchableOpacity} from 'react-native';
import {Text, Icon, Divider} from '@ui-kitten/components';

interface SettingsItemProps {
  label: string;
  onPress?: () => void;
  icon?: string;
  children?: React.ReactNode;
}

export const SettingsMenuItem: React.FC<SettingsItemProps> = ({
  label = '',
  onPress,
  icon,
  children,
}) => {
  const [toggle, setToggle] = React.useState(false);

  let rightIcon = children ? 'chevron-right-outline' : icon ? icon : null;

  const onPressHandler = () => {
    if (children) {
      setToggle(prev => !prev);
    } else {
      onPress?.();
    }
  };

  return (
    <View testID="menuItem">
      <TouchableOpacity onPress={onPressHandler}>
        <View style={styles.menuItem}>
          <Text>{label}</Text>
          {rightIcon && (
            <Icon
              style={styles.icon}
              fill="#8F9BB3"
              name={rightIcon}
              testID="menu_icon"
            />
          )}
        </View>
        <Divider />
      </TouchableOpacity>
      {children && toggle && (
        <View testID="menuItemChild" style={styles.childrenWrp}>
          {children}
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  list: {
    backgroundColor: 'cyan',
  },
  menuItem: {
    paddingVertical: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  menuText: {
    fontSize: 14,
  },
  icon: {
    width: 24,
    height: 24,
  },
  childrenWrp: {
    paddingLeft: 10,
  },
});
