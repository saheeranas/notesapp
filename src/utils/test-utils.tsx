import * as React from 'react';
import {render} from '@testing-library/react-native';
import * as eva from '@eva-design/eva';
import {ApplicationProvider, IconRegistry} from '@ui-kitten/components';
import {EvaIconsPack} from '@ui-kitten/eva-icons';

// Navigation
// import RootNavigation from '../routes/RootNavigation';

const AllTheProviders = ({children}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={eva.light}>
        {children}
      </ApplicationProvider>
    </>
  );
};

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
