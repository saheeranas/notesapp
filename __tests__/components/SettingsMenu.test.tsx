import 'react-native';
import React from 'react';
import {render, fireEvent} from '../../src/utils/test-utils';

import {SettingsMenuItem} from '../../src/components/SettingsMenu';

it('renders SettingsMenuItem', () => {
  const tree = render(<SettingsMenuItem label="" />, {}).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with a label', () => {
  const {getByText} = render(<SettingsMenuItem label="Dummy label" />, {});

  getByText('Dummy label');
});

it('renders with an icon', () => {
  const {getByTestId} = render(
    <SettingsMenuItem label="Dummy label" icon="star-outline" />,
    {},
  );

  expect(getByTestId('menu_icon')).toBeVisible();
});

it('onPress fires menu item which have no children', () => {
  let onPressMock = jest.fn();
  const {getByText} = render(
    <SettingsMenuItem label="Dummy label" onPress={onPressMock} />,
    {},
  );

  fireEvent.press(getByText('Dummy label'));
  expect(onPressMock).toHaveBeenCalledTimes(1);
});

// describe('Menu Item with children', () => {
//   it('renders with children', () => {
//     const {getByTestId, getByText} = render(
//       <SettingsMenuItem label="Parent Item">
//         <SettingsMenuItem label="Child 1" />
//         <SettingsMenuItem label="Child 2" />
//       </SettingsMenuItem>,
//       {},
//     );

//     const parent = getByTestId('menuItem');

//     fireEvent.press(parent);

//     expect(getByText('Child 1')).toBeVisible();
//   });
// });
