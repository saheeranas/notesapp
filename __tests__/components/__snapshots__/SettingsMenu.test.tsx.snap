// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SettingsMenuItem 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    testID="menuItem"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 10,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        appearance="default"
        style={
          Array [
            Object {
              "backgroundColor": "#EDF1F7",
              "height": 1,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
